libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/snap/SnapToRoute.java

```diff
@@ -2,124 +2,28 @@
 
 import android.location.Location;
 import android.support.annotation.NonNull;
-import android.support.annotation.Nullable;
 
-import com.mapbox.geojson.Feature;
-import com.mapbox.geojson.LineString;
-import com.mapbox.geojson.Point;
-import com.mapbox.services.android.navigation.v5.routeprogress.RouteLegProgress;
+import com.mapbox.navigator.NavigationStatus;
 import com.mapbox.services.android.navigation.v5.routeprogress.RouteProgress;
-import com.mapbox.services.android.navigation.v5.routeprogress.RouteStepProgress;
-import com.mapbox.services.android.navigation.v5.utils.MathUtils;
-import com.mapbox.turf.TurfConstants;
-import com.mapbox.turf.TurfMeasurement;
-import com.mapbox.turf.TurfMisc;
 
-import java.util.List;
-
-import static com.mapbox.core.constants.Constants.PRECISION_6;
-
-/**
- * This attempts to snap the user to the closest position along the route. Prior to snapping the
- * user, their location's checked to ensure that the user didn't veer off-route. If your application
- * uses the Mapbox Map SDK, querying the map and snapping the user to the road grid might be a
- * better solution.
- *
- * @since 0.4.0
- */
 public class SnapToRoute extends Snap {
 
   @Override
   public Location getSnappedLocation(Location location, RouteProgress routeProgress) {
-    Location snappedLocation = snapLocationLatLng(location, routeProgress.currentStepPoints());
-    snappedLocation.setBearing(snapLocationBearing(routeProgress));
-    return snappedLocation;
+    // No impl
```

Comment:
why? Is the location already snapped?

Reply:
`getSnappedLocationWith(Location location, NavigationStatus status)` is what is now used in `SnapToRoute` to retrieve the snapped location.  We discussed that this isn't ideal as we are basically ignoring the abstraction requirements - let's continue to iterate, as I know @Guardiola31337  had some thoughts on how to improve this as well